src/main/java/org/reflections/vfs/Vfs.java

```diff
@@ -205,7 +205,7 @@ public boolean apply(File file) {
     public static enum DefaultUrlTypes implements UrlType {
         jarFile {
             public boolean matches(URL url) {
-                return url.getProtocol().equals("file") && url.toExternalForm().contains(".jar");
+                return url.getProtocol().equals("file") && url.toExternalForm().endsWith(".jar");
```

Comment:
Aren't url's like this too?
  file://some/path/my.jar!org/foo/bar
The ! is often used in java url's to indicate that it's inside zip (not sure if it's used in these kind though).

I agree contains has false positives (like tony.jarvis) but endswith might have false negatives (like my.jar!org).